### PR TITLE
🚩 Upgrade to NextJS v15

### PR DESCRIPTION
**Pull request by shubhamkhan**
### Application NextJS Upgrade to v15
This pull request upgrades the Next.js framework from the previous version to v15.
Please review the changes, and let me know if you have any questions or feedback.

### CODE REVIEW

**Review by wei — CHANGES_REQUESTED** [review]
Please follow the full NextJS 15 upgrade guide which includes upgrading to React 19.

Please review all the framework features and make adjustments to the app as needed.

Please recommend future improvements that can be done after the initial upgrade to NextJS 15.

**Comment by shubhamkhan** [comment]
Hi @wei, I have upgrade to React v19 with dependencies.

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "license": "MIT",
   "repository": "https://github.com/wei/socialify.git",
   "scripts": {
-    "dev": "next dev",
+    "dev": "next dev --turbopack",
```
Comment (wei) [inline package.json:9]: can you explain this change or link to the source?

File: package.json [hunk 2]
```diff
@@ -71,5 +71,5 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
-  }
+  }
```
Comment (wei) [inline package.json:74]: Please run linter before committing.

**Review by wei — CHANGES_REQUESTED** [review]
Could you comment on any new features or changes nextjs and react 19 brings that we should adopt or could benefit from in the project? Thank you!

btw also missing changeset file as seen in the first comment by changeset bot.

File: package.json (hunk 1 above)
Comment (shubhamkhan) [reply package.json:9]: Turbopack I have used for incremental bundler optimized for JavaScript and TypeScript in Next.js and faster local development

**Comment by shubhamkhan** [comment]
Hi @wei,
- Added a changeset to document the updates.
- Ensured code consistency with linting and ran tests successfully.

**Comment by wei** [comment]
Hello, thanks for the update, however, we're still missing these items from the last review:

> Please review all the framework features and make adjustments to the app as needed.
> 
> Please recommend future improvements that can be adopted for NextJS after the initial upgrade.
> 
> Could you comment on any new features or changes nextjs and react 19 brings that we should adopt or could benefit from in the project? Thank you!

Please click re-request review once these points have been addressed.

File: package.json (hunk 1 above)
Comment (wei) [reply package.json:9]: Thanks I see the blogpost at https://nextjs.org/blog/turbopack-for-development-stable is using `--turbo` but your change is `--turbopack`. Do you have documentation on the difference?

**Review by wei — CHANGES_REQUESTED** [review]
Please address previous outstanding comments. Thank you!

File: package.json (hunk 1 above)
Comment (wei) [reply package.json:9]: @shubhamkhan could you respond to my comment above?

File: package.json (hunk 2 above)
Comment (wei) [reply package.json:74]: Hey @shubhamkhan  This is not yet resolved ^

**Comment by wei** [comment]
Great! Please check outstanding comments above (I tagged you again in case you missed it)

Please list out all the changes you have implemented with source links so we can get them fully reviewed and approved. 

Thank you!

File: src/components/mainWrapper.tsx [hunk 3]
```diff
@@ -23,7 +23,11 @@ const MainWrapper = ({ response }: MainWrapperProps) => {
   }
 
   useEffect(() => {
-    if (!response || !response.repository) {
+    if (!response) {
+      return
+    }
```
Comment (wei) [inline src/components/mainWrapper.tsx:28]: Any reason to no longer redirect when response is falsy?

File: src/components/toaster.tsx [hunk 4]
```diff
@@ -19,7 +19,12 @@ const _helper = (type: keyof typeof ToastTypeMap) => {
   return (message: string) =>
     hotToast.custom((_t) => {
       return (
-        <div className={`alert ${className} w-fit shadow-lg`}>
+        <div
+          className={`alert ${className} w-fit shadow-lg`}
+          role="alert"
+          aria-live="polite"
+          aria-atomic="true"
```
Comment (wei) [inline src/components/toaster.tsx:26]: What are these for? Could you link me the sources?

File: pages/index.tsx [hunk 5]
```diff
@@ -1,7 +1,11 @@
 import Repo from '../src/components/repo/repo'
 
 const HomePage = () => {
-  return <Repo />
+  return (
+    <>
+      <Repo />
+    </>
+  )
```
Comment (wei) [inline pages/index.tsx:8]: Any reason for this change?

File: pages/api/graphql.ts [hunk 6]
```diff
@@ -20,6 +20,7 @@ const graphQLEndpoint = async (req: NextRequest) => {
       'content-type': 'application/json',
     },
     body: req.body,
+    cache: 'force-cache',
```
Comment (wei) [inline pages/api/graphql.ts:23]: How does cache behave with POST requests? Could you help me understand? TY!

File: pages/index.tsx (hunk 5 above)
Comment (shubhamkhan) [reply pages/index.tsx:8]: For maintaining coding structure

**Review by shubhamkhan — COMMENTED** [review]
Hi @wei, Can you review the updated code and validate, I run lint and test before commit. Thank You!

**Comment by wei** [comment]
👋  Hey hey @shubhamkhan thanks for all the help thus far. I have gone ahead and upgraded a ton of other dependencies in master. I'm just wondering if you'll have some time to carry this PR through the finish line 😄 

Just to summarize a few requested changes:

- [ ] Rebase / merge in upstream master branch to get the latest changes and resolve merge conflicts.
- [ ] Remove irrelevant dependency and code changes, keep this PR single purposed - Next.js 15 and React 19 upgrade only
- [ ] Double check the latest versions on npm registry and make sure we are using the latest
- [ ] Please review all the framework features and make adjustments to the app as needed
- [ ] Please provide a description of every change for review
- [ ] Run linter

Afterwards, if you have time, 

- [ ] Please recommend future improvements that can be adopted for NextJS after the initial upgrade.
- [ ] Could you comment on any new features or changes nextjs and react 19 brings that we should adopt or could benefit from in the project? 

Please let me know, completely fine if you don't currently have the bandwidth to tackle this, many thanks!

**Review by wei — CHANGES_REQUESTED** [review]
👋  Hey hey @shubhamkhan thanks for all the help thus far. I have gone ahead and upgraded a ton of other dependencies in master. I'm just wondering if you'll have some time to carry this PR through the finish line 😄 

Just to summarize a few requested changes:

- [ ] Rebase / merge in upstream master branch to get the latest changes and resolve merge conflicts.
- [ ] Remove irrelevant dependency and code changes, keep this PR single purposed - Next.js 15 and React 19 upgrade only
- [ ] Double check the latest versions on npm registry and make sure we are using the latest
- [ ] Please review all the framework features and make adjustments to the app as needed
- [ ] Please provide a description of every change for review
- [ ] Run linter

Afterwards, if you have time, 

- [ ] Please recommend future improvements that can be adopted for NextJS after the initial upgrade.
- [ ] Could you comment on any new features or changes nextjs and react 19 brings that we should adopt or could benefit from in the project? 

Please let me know, completely fine if you don't currently have the bandwidth to tackle this, many thanks!

**Comment by KemingHe** [comment]
Hi @wei , I'm happy to continue and ultimately deliver this upgrade building on top of what @shubhamkhan have so far. Let me know how you'd like me to proceed after #405 . 

Best, K.H

**Comment by wei** [comment]
> Hi @wei , I'm happy to continue and ultimately deliver this upgrade building on top of what @shubhamkhan have so far. Let me know how you'd like me to proceed after #405 .
> 
> Best, K.H

This is the last major remaining roadblock before new features can be worked on. @KemingHe Your help would be greatly appreciated if you've got time to help take this across the finish line. Feel free to make a new branch/PR as I also don't seem to have write permission on this one. 

@shubhamkhan Please let me know if you will have time to help with this before I close this PR due to inactivity. TY!

**Comment by shubhamkhan** [comment]
Hi @wei and @KemingHe Thanks for continuing the PR.

**Pull request closed without merging**